docs/src/main/asciidoc/grpc-service-implementation.adoc

```diff
@@ -0,0 +1,242 @@
+        // Just returns a stream emitting an item every 2ms and stopping after 10 items.
+        return Multi.createFrom().ticks().every(Duration.ofMillis(2))
+                .transform().byTakingFirstItems(10)
+                .map(l -> Item.newBuilder().setValue(Long.toString(l)).build());
+    }
+
+    @Override
+    public Uni<Empty> sink(Multi<Item> request) {
+        // Reads the incoming streams, consume all the items.
+        return request
+                .map(Item::getValue)
+                .map(Long::parseLong)
+                .collectItems().last()
+                .map(l -> Empty.newBuilder().build());
+    }
+
+    @Override
+    public Multi<Item> pipe(Multi<Item> request) {
+        // Reads the incoming stream, compute a sum and return the cumulative results
+        // in the outbound stream.
+        return request
+                .map(Item::getValue)
+                .map(Long::parseLong)
+                .onItem().scan(() -> 0L, Long::sum)
+                .onItem().apply(l -> Item.newBuilder().setValue(Long.toString(l)).build());
+    }
+}
+----
+
+== Health check
+For the exposed services, Quarkus gRPC exposes health information in the following format:
+[source,protobuf]
+----
+syntax = "proto3";
+
+package grpc.health.v1;
+
+message HealthCheckRequest {
+  string service = 1;
+}
+
+message HealthCheckResponse {
+  enum ServingStatus {
+    UNKNOWN = 0;
+    SERVING = 1;
+    NOT_SERVING = 2;
+  }
+  ServingStatus status = 1;
+}
+
+service Health {
+  rpc Check(HealthCheckRequest) returns (HealthCheckResponse);
+
+  rpc Watch(HealthCheckRequest) returns (stream HealthCheckResponse);
+}
+----
+
+Clients can specify the fully qualified service name to get the health status of a specific service
+or skip specifying the service name to get the general status of the gRPC server.
+
+For more details, check out the
+https://github.com/grpc/grpc/blob/v1.28.1/doc/health-checking.md[gRPC documentation]
+
+Additionally, if Quarkus SmallRye Health is added to the application, a readiness check for
+the state of the gRPC services will be added to the MicroProfile Health endpoint response, that is `/health`.
+
+== Reflection Service
+
+Quarkus gRPC Server implements the https://github.com/grpc/grpc/blob/master/doc/server-reflection.md[reflection service].
+This service allows tools like https://github.com/fullstorydev/grpcurl[grpcurl] or https://github.com/gusaul/grpcox[grpcox] to interact with your services.
+
+The reflection service is enabled by default in _dev_ mode.
+In test or production mode, you need to enable it explicitly by setting `quarkus.grpc-server.enable-reflection-service` to `true`.
+
+== Server configuration
+
+By default, the server is exposed on `localhost:9000` and uses `plain-text`.
+
+The following configuration properties are supported:
+
+|===
+|Property |Description |Default Value
+
+|quarkus.grpc-server.enable-reflection-service | Enables the reflection service  | `false` (`true` in dev mode)
```

Comment:
This should use our automatic doc generation.

Reply:
Yes, I need to try to understand why it was not working from the other repo. Maybe now it is going to work.

Reply:
It was not working for external repos, @machi1990 worked on fixing that but having it in the main repo should make it work anyway. If not, please ping @machi1990 and me, we will help!

Reply:
Fixed!